src/main/java/com/fvv/bookstore/view/EmployeeViewImpl.java

```diff
@@ -117,6 +119,53 @@ public void removeEmployee() {
 		}
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void listEmployeesByName() {
+		try {
+			StringBuilder sb = new StringBuilder();
+			String searchName = JOptionPane.showInputDialog("Insert a name to find");
+			List<Employee> employees = this.employeeController.listEmployeesByName(searchName);
+			if(employees != null && !employees.isEmpty()) {
+				for(Employee emp : employees) {
+					sb.append(emp).append(Constants.LINE_SEPARATOR);
+				}
+				JOptionPane.showMessageDialog(null, sb.toString(), "Listing the Employees", 
+						JOptionPane.PLAIN_MESSAGE);
+			} else {
+				JOptionPane.showMessageDialog(null, "There are no items to show!");
+			}			
+		} catch (ControllerException | PersonNotFoundException e) {
+			e.printStackTrace();
+			JOptionPane.showMessageDialog(null, e.getMessage(), "Error", 
+					JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void getSalaryWithCommission() {
+		try {
+			Long searchId = Long.parseLong(JOptionPane.showInputDialog("Insert the employee ID"));
+			Employee employee = this.employeeController.findEmployee(searchId);
+			employee.setId(searchId);
+			
+			final Double percentage = Double.parseDouble(JOptionPane.showInputDialog("Input the percentage in decimal"));
+			final Date date = DateUtil.stringToDate(JOptionPane.showInputDialog("Insert the date"));
+			final Double newSalary = this.employeeController.getSalaryWithCommission(percentage, employee, date);
+			
+			JOptionPane.showMessageDialog(null, "Employee: " + employee.getId() + " - " + employee.getName() + " - Salary with Commission: " + newSalary);
```

Comment:
you dont need the word "Employee:" in the message, it would be better:

Month: MM
ID: XXX
Name: YYYY
Current Salary: R$ 0.000,00
Commission Earned: R$ 0.000,00
Total Salary: R$ 0.000,00

Reply:
ok